In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats


# Trabajo 2

La base de datos con la que le corresponde trabajar, se obtiene como una muestra aleatoria que corresponde a corresponde a los resultados obtenidos por los jóvenes de todo el país en las pruebas Saber 11, del año 2022. La información aparece agregada por Colegio y solo abarca los colegios de Antioquia. Dicha base contiene las variables: NATURALEZA (Colegio Oficial o No-Oficial), JORNADA (Asociado a los diferentes tipos de jornada de estudio), PROMLECT (Promedio obtenido por el colegio en Lectura Crítica) y PROMMATE (Promedio obtenido por el colegio en Matemáticas).

In [2]:
# Leer archivo csv

raw_data_url = "https://raw.githubusercontent.com/repos-especializacion-UdeA/estadistica/refs/heads/main/trabajo2/DatosTrabajo2EAE20242.csv"

# Leer el archivo CSV
df = pd.read_csv(raw_data_url)

# Mostrar las primeras filas del DataFrame
df.head()

NATURALEZA   JORNADA  PROMLECT  PROMMATE
0    OFICIAL  COMPLETA        33        18
1    OFICIAL  COMPLETA         0        19
2    OFICIAL  COMPLETA        30        21
3    OFICIAL  COMPLETA        44        21
4    OFICIAL  COMPLETA        31        25

## Funciones resumen

In [3]:
df.shape

(1807, 4)

In [4]:
df.describe()

PROMLECT     PROMMATE
count  1807.000000  1807.000000
mean     50.321527    47.105147
std       7.028032     7.897110
min       0.000000    18.000000
25%      45.000000    41.000000
50%      50.000000    47.000000
75%      55.000000    52.000000
max      74.000000    76.000000

In [5]:
df.value_counts()

NATURALEZA  JORNADA        PROMLECT  PROMMATE
OFICIAL     UNICA          53        49          10
            COMPLETA       44        41          10
            MANANA         56        52           9
NO OFICIAL  COMPLETA       63        63           9
OFICIAL     FIN DE SEMANA  43        39           9
                                                 ..
            COMPLETA       44        30           1
                                     34           1
                                     44           1
                           45        29           1
            UNICA          65        64           1
Length: 964, dtype: int64

In [6]:
df.isnull().sum()

NATURALEZA    0
JORNADA       0
PROMLECT      0
PROMMATE      0
dtype: int64

In [7]:
print(df.NATURALEZA.unique())
print(df.JORNADA.unique())

['OFICIAL' 'NO OFICIAL']
['COMPLETA' 'MANANA' 'UNICA' 'NOCTURNA' 'FIN DE SEMANA' 'TARDE']


## Punto 1 - Intervalo de confianza

Construir en Python un IC para al 95% de confianza necesario para responder: ¿Puede afirmarse que el resultado medio obtenido por los colegios en Lectura es superior a 45 puntos? Justifique su respuesta.

### Solución

Sea **$X_i$** la VA relacionada con el promedio obtenido en la prueba de lectura critica por el i-esimo colegio 

A partir de la tabla podemos obtener los siguientes valores:

In [13]:
# Datos del problema
data_prom_lectura = df.PROMLECT.copy()


# Calculo de la media y la desviacion estandar muestral
n = data_prom_lectura.shape[0]
X = data_prom_lectura.mean()
S = data_prom_lectura.std()
# Despliegue de los resultados
print(f"n = {n}")
print(f"X = {X:.5f}")
print(f"S = {S:.5f}")


n = 1807
X = 50.32153
S = 7.02803


'\nn = df.shape[0]\nX = df.PROMLECT.mean()\nS = df.PROMLECT.std()\nprint(f"n = {n}")\nprint(f"X = {X:.5f}")\nprint(f"S = {S:.5f}")\n\nNC = 0.95 # Nivel de confianza\nalpha = 1 - NC \n'

Como la poblacion no es normal, el numero de datos que esta tiene es de mas de 30 y la varianza es desconocida, usamos la siguiente formula para el intervalo de confianza:

$$
IC = \overline{X}\pm Z_{\alpha/2}\frac{S}{\sqrt{n}}
$$

Inicialmente vamos a calcular para $\alpha = 0.05$:

$
Z_{\alpha/2} = Z_{0.025}  
$

In [19]:
NC = 95 # Nivel de confianza
alpha = 1 - NC/100 

Z = stats.norm().ppf(1 - alpha/2)
print(f"Z = {Z:.5f}")

EBM = Z*S/np.sqrt(n)
print(f"EBM = {EBM:.5f}")

IC_left = X - EBM 
IC_right = X + EBM
print(f"({X:.5f} - {EBM:.5f},{X:.5f} + {EBM:.5f})= ({IC_left:.5f},{IC_right:.5f})")

Z = 1.95996
EBM = 0.32404
(50.32153 - 0.32404,50.32153 + 0.32404)= (49.99748,50.64557)


**Respuesta numerica**: Como el intervalo en el cual se encuentra la media en los para lectura esta en el rango $(49.99748,50.64557)$, se puede afirmar con un nivel de confianza del 95% que el resultado promedio en la prueba de lectura es superior a 45 puntos.

Como el límite inferior del intervalo de confianza es 49.997, que es mayor que 45, podemos afirmar con un 95% de confianza que el promedio de los resultados en Lectura es superior a 45 puntos. El valor de 45 puntos no está dentro del intervalo de confianza. Dado que todo el intervalo está por encima de 45 puntos, podemos concluir que es muy poco probable que el promedio verdadero sea igual o menor a 45 puntos. Por lo tanto, podemos afirmar que el promedio es superior a 45 puntos con un alto nivel de confianza.

## Punto 2 - Prueba de hipotesis

¿Se puede afirmar que el resultado medio obtenido por los colegios en Matemáticas es mayor en los Privados que en los Públicos? Justifique su respuesta mediante una prueba de hipótesis con una significancia del 4%.

In [52]:
# Informacion de la prueba de matematicas 
# publicos
df_publico = df[df['NATURALEZA'] == 'OFICIAL']
print(df_publico.shape)
# privados
df_privado = df[df['NATURALEZA'] == 'NO OFICIAL']
print(df_privado.shape)

# Nivel de significancia
alpha = 0.04 # 4%


(1466, 4)
(341, 4)


In [53]:
notas_mat_privado = df_privado.PROMMATE
notas_mat_publico = df_publico.PROMMATE

Se tienen las siguientes VA:
* Sea **$X_i$** la VA relacionada con el promedio obtenido en la prueba de matematicas por el i-esimo colegio privado
* Sea **$Y_i$** la VA relacionada con el promedio obtenido en la prueba de matematicas por el j-esimo colegio publico

Tenemos los siguientes datos por colegio:
* **Colegio privado**:
  * $1466$
* **Colegio publico**:

In [54]:
# Colegio privado
nx = notas_mat_privado.shape[0]
X = notas_mat_privado.mean()
Sx = notas_mat_privado.std()
print(f"nx = {nx}")
print(f"X = {X:.5f}")
print(f"Sx = {Sx:.5f}")

nx = 341
X = 52.75367
Sx = 9.59462


In [55]:
# Colegio publico
ny = notas_mat_publico.shape[0]
Y = notas_mat_publico.mean()
Sy = notas_mat_publico.std()
print(f"ny = {ny}")
print(f"Y = {Y:.5f}")
print(f"Sy = {Sy:.5f}")

ny = 1466
Y = 45.79127
Sy = 6.80895


Hipotesis: Enunciado: $\mu_x > \mu_y$ es decir $\mu_x - \mu_y > 0$

* $H_o: \mu_x - \mu_y \leq 0$
* $H_a: \mu_x - \mu_y > 0$

Calculo del estadistico de prueba

In [56]:
d0 = 0
Zc = (X - Y - d0) / ((Sx**2 / nx + Sy**2 / ny)**0.5)
print(Zc)

12.678071514420095


In [57]:
VP = 1 - stats.norm.cdf(Zc)
print(VP)

0.0


In [58]:
# Paso 4 - Si VP <= a, se rechaza la hipotesis nula, para un a = 0.05
print(VP >= alpha)

False


In [51]:
# se rechaza la hipotesis nula, 

Este valor de 𝑧 z es muy alto, lo que indica una diferencia significativa entre las medias de los colegios privados y públicos en cuanto a los resultados en Matemáticas.

Como el valor de  𝑧 z es mucho mayor que los valores críticos típicos para un nivel de significancia de 4% (que es aproximadamente 1.75 para una prueba unilateral), podemos rechazar la hipótesis nula con confianza, lo que confirma que el resultado medio en Matemáticas es mayor en los colegios privados que en los públicos.

* https://github.com/OpenIntroStat/openintro-statistics-slides
* https://365datascience.com/calculators/confidence-interval-calculator/
* https://oercommons.org/authoring/18167-openstax-introductory-statistics-videos-with-lectu/view
* https://stanford.edu/~shervine/teaching/cme-106/
* https://datatab.es/tutorial/confidence-interval
* https://neuraldatascience.io/3b-ai_assisted/multi_data_files_2.html
* https://stanford.edu/~shervine/teaching/cme-106/cheatsheet-statistics#confidence-intervals